chapter/1/rpc-reviewed-dont-keep.md

```diff
@@ -0,0 +1,619 @@
+parties. However, in Cap'n Proto "the RPC layer is not robust against resource
+exhaustion attacks, possibly allowing denials of service", nor has it undergone
+any formal verification {% cite capnprotosecure --file rpc %}.
+
+Although, it's possible to come up with a *Threat Model* that would make an RPC
+implementation insecure to use, however, one has to understand that using any
+distributed system increases the attack surface anyways and claiming one
+*paradigm* to be more secure than another would be a biased statement, since
+*paradigms* are generally an idea and it depends on different system designers
+to use these *paradigms* to build their systems and take care of features
+specific to real systems, like security and load-balancing. There's always a
+possibility of rerouting a request to a malicious server(if the registry gets
+hacked), or there's no trust between the *caller* and *callee*. However, we
+maintain that RPC *paradigm* is not secure or insecure(for that matter), and
+that the most secure systems are the ones that are in an isolated environment,
+disconnected from the public internet with a self-destruct mechanism{% cite self
+--file rpc %} in place, in an impenetrable bunker, and guarded by the Knights
+Templar(*they don't exist! Well, maybe Fort Meade comes close*).
+
+## Discussion:
+
+RPC *paradigm* shines the most in *request-response* mechanisms. Futures and
+Promises also appear to a new breed of RPC. This leads one to question, as to
+whether every *request-response* system is a modified implementation to of the
+RPC *paradigm*, or does it actually bring anything new to the table? These
+modern communication protocols, like HTTP and REST, might just be a different
+flavor of RPC. In HTTP, a client *requests* a web page(or some other content),
+the server then *responds* with the required content. The dynamics of this
+communication might be slightly different from your traditional RPC, however, an
+HTTP Stateless server adheres to most of the concepts behind RPC *paradigm*.
+Similarly, consider sending a request to your favorite Google API. Say, you want
+to translate your latitude/longitude to an address using their Reverse Geocoding
+API, or maybe want to find out a good restaurant in your vicinity using their
+Places API, you'll send a *request* to their server to perform a *procedure*
+that would take a few input arguments, like the coordinates, and return the
+result. Even though these APIs follow a RESTful design, it appears to be an
+extension to the RPC *paradigm*.
+
+RPC paradigm has evolved over time. It has evolved to the extent that,
+currently, it's become very difficult differentiate RPC from non-RPC. For the
+past decades, researchers and industry leaders have tried to come up with
+*their* definition of RPC. The proponents of RPC paradigm view every
+*request-response* communication as an implementation the RPC paradigm while
+those against RPC try to explicitly come up with the bounds of RPC. RPC
```

Comment:
Maybe not the _bounds_ per se, but they enumerate the limitations which seem to change with each passing decade.